src/Bridge/Symfony/Bundle/Command/DebugResourceCommand.php

```diff
@@ -0,0 +1,122 @@
+            ->addArgument('class', InputArgument::REQUIRED, 'The class you want to debug');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $resourceClass = $input->getArgument('class');
+
+        $resourceCollection = $this->resourceMetadataCollectionFactory->create($resourceClass);
+
+        if (0 === \count($resourceCollection)) {
+            $output->writeln(sprintf('<error>No resources found for class %s</error>', $resourceClass));
+
+            return Command::INVALID;
+        }
+
+        $shortName = (false !== $pos = strrpos($resourceClass, '\\')) ? substr($resourceClass, $pos + 1) : $resourceClass;
+
+        $helper = $this->getHelper('question');
+
+        $resources = [];
+        foreach ($resourceCollection as $resource) {
+            if ($resource->getUriTemplate()) {
+                $resources[] = $resource->getUriTemplate();
+                continue;
+            }
+
+            foreach ($resource->getOperations() as $operation) {
+                if ($operation->getUriTemplate()) {
+                    $resources[] = $operation->getUriTemplate();
+                    break;
+                }
+            }
+        }
+
+        if (\count($resourceCollection) > 1) {
+            $questionResource = new ChoiceQuestion(
+                sprintf('There are %d resources declared on the class %s, which one do you want to debug ? ', \count($resourceCollection), $shortName).\PHP_EOL,
+                $resources
+            );
+
+            $answerResource = $helper->ask($input, $output, $questionResource);
+            $resourceIndex = array_search($answerResource, $resources, true);
+            $selectedResource = $resourceCollection[$resourceIndex];
+        } else {
+            $selectedResource = $resourceCollection[0];
+            $output->writeln(sprintf('Class %s declares 1 resource.', $shortName).\PHP_EOL);
+        }
+
+        $operations = ['Debug the resource itself'];
+        foreach ($selectedResource->getOperations() as $operationName => $operation) {
+            $operations[] = $operationName;
+        }
+
+        $questionOperation = new ChoiceQuestion(
+            sprintf('There are %d operation%s declared on the resource, which one do you want to debug ? ', $selectedResource->getOperations()->count(), $selectedResource->getOperations()->count() > 1 ? 's' : '').\PHP_EOL,
+            $operations
+        );
+
+        $answerOperation = $helper->ask($input, $output, $questionOperation);
+        if ('Debug the resource itself' === $answerOperation) {
+            $this->dumper->dump($this->cloner->cloneVar($selectedResource));
+            $output->writeln('Successfully dumped the selected resource');
+
+            return Command::SUCCESS;
+        }
+
+        $this->dumper->dump($this->cloner->cloneVar($resourceCollection->getOperation($answerOperation)));
```

Comment:
Why do we need to clone the object?

Reply:
Because `dump` function needs `Symfony\Component\VarDumper\Cloner\Data` as argument